signac/contrib/project.py

```diff
@@ -1196,16 +1214,18 @@ def write_statepoints(self, statepoints=None, fn=None, indent=2):
         with open(fn, "w") as file:
             file.write(json.dumps(tmp, indent=indent))
 
-    def _register(self, job):
+    def _register(self, _id, statepoint):
```

Comment:
I changed this method to reduce overhead. The id and state point are already known in all the cases where `project._register` is called, but the previous implementation had to explicitly perform a conversion back from the synced state point to a plain dict. That accounted for some of the slowdown between `master` and `feature/lazy-statepoint-loading` in #239 when opening jobs by state point.